src/s3_plugin.js

```diff
@@ -93,14 +96,22 @@ module.exports = class S3Plugin {
       }
 
       if (isDirectoryUpload) {
-        let dPath = this.addSeperatorToPath(this.options.directory)
-        this.getAllFilesRecursive(dPath)
-          .then(this.filterAndTranslatePathFromFiles(dPath))
-          .then(this.filterAllowedFiles.bind(this))
-          .then(this.uploadFiles.bind(this))
-          .then(this.changeHtmlUrls.bind(this))
-          .then(this.invalidateCloudfront.bind(this))
-          .then(() => cb())
+        this.addGitHashToBasePath(this.options.basePath)
```

Comment:
I really want to keep the promise chain flat over here. If you follow my comments below you would be able to keep the promise queue flat and just move the assignment of `dPath` to outside the promise. Lastly this still needs to be an option if it is not a directory upload.